### PR TITLE
feat: OpenAPI extract schema from Object or Array

### PR DESCRIPTION
**Pull request by Cobalt0s**
# Background

OpenAPI tool using response object schema, locates the field that holds an array and uses that schema to describe object that we return during Read/ListObjectMetadata.

# Changes

* Response schema might already be a an array, so there is no need for this extra "locate" step. We need to extract fields from current `array-item schema`.
* `ObjectCheck` was renamed to `ObjectArrayLocator`. Array holder would use this callback to find field name, ex (product schema is located under products response field): 
```json
{
  "field1":[{}, {}, {}],
  "field2":{...},
  "products":[],
  "field3":{},
}
```
A direct return of `[ {product1},  {product2},  {product3}, ]` doesn't need `ObjectArrayLocator`.

# TLDR
Array of objects is either returned directly or part of top level object under some field name resolved by `ObjectArrayLocator`.
Added support to arrays as top level JSON node.
image

### CODE REVIEW

**Comment by Cobalt0s** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/1109?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#1111** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/1111?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1110** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/1110?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1109** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/1109?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#1108** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/1108?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1107** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/1107?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Cobalt0s and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by eberle1080 — APPROVED** [review]
